### PR TITLE
Eos 23372 : Design for  Fix Key size, Variable Value size node type

### PR DESCRIPTION
**Pull request by RadhaGulhane13**

### CODE REVIEW

File: btree/btree.c [hunk 1]
```diff
@@ -3085,6 +3085,594 @@ static void ff_rec_del_credit(const struct nd *node, m0_bcount_t ksize,
...
+ */
+
+/**
+ *
+ * Proposed Node Structure for Fix Sized Keys and Variable Sized Value :
+ *
+ * +-----------+----+-+----+-+----+-+------------------------------------+----+----+----+
+ * |           |    | |    | |    | |                                    |    |    |    |
+ * |           |    | |    | |    | |                                    |    |    |    |
+ * |Node Header| K0 | | K1 | | K2 | | ----->                      <----- | V0 | V1 | V2 |
```
Comment (shashank-parulekar) [inline btree/btree.c:3103]: The Values will be in the reverse order from the way it is shown here just like how Yash has it in his design.

Comment (RadhaGulhane13) [reply btree/btree.c:3103]: Yes right, it will be v2 ,v1 , v0, it is my mistake

**Review by NikhilBirgade — APPROVED** [review]

File: btree/btree.c [hunk 2]
```diff
@@ -3085,6 +3085,594 @@ static void ff_rec_del_credit(const struct nd *node, m0_bcount_t ksize,
...
+ * offset for value at index i and offset for previous value.
+ *
+ * In this way, keys (which will also include offset for value) will be added
+ * from right to left starting after node header region and value will be added
+ * from left to right starting from end of the node.
+ *
+ */
+struct fkvv_head {
+	struct m0_format_header  fkvv_fmt;    /*< Node Header */
+	struct node_header       fkvv_seg;    /*< Node type information */
```
Comment (ydb242) [inline btree/btree.c:3144]: Kindly add this comment as it has been added to ff and vkvv head structures.
	`
	/**
	 * The above 2 structures should always be together with node_header
	 * following the m0_format_header.
	 */
`

**Review by ydb242 — APPROVED** [review]

**Review by shashank-parulekar — CHANGES_REQUESTED** [review]

File: btree/btree.c [hunk 3]
```diff
@@ -3085,6 +3085,594 @@ static void ff_rec_del_credit(const struct nd *node, m0_bcount_t ksize,
...
+ * The above structure represents the way fixed key size and variable value size
+ * node format will get stored in memory.
+ *
+ * Node can be mainly divided into three region, node header, region for keys
+ * and region for values.
+ *
+ * Node header will contain all the relevant information about the node
+ * including node type.
+ *
+ * The key region will be start after the node header region. Keys will be added
```
[the remark this answers is not in the document]
Comment (RadhaGulhane13) [reply btree/btree.c]: done.

File: btree/btree.c [hunk 4]
```diff
@@ -3085,6 +3085,594 @@ static void ff_rec_del_credit(const struct nd *node, m0_bcount_t ksize,
...
+ * (from the end of node header region) in the sorted order and each key will be
+ * associated with offset which will be the offset for the value. The offset for
```
[the remark this answers is not in the document]
Comment (RadhaGulhane13) [reply btree/btree.c]: updated.

**Review by shashank-parulekar — APPROVED** [review]

**Pull request merged**